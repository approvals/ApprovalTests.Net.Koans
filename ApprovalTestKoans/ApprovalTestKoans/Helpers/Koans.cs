namespace ApprovalTestKoans.Helpers
{
	public class Koans
	{
		public string ___ = "Please fill in the blank";
		public int ____ = 0;

//  public String decode(String string)
//  {
//    int offshift = 9;
//    return push(string, offshift);
//  }
//  public String push(String string, int offshift)
//  {
//    StringBuffer b = new StringBuffer();
//    for (char c : string.toCharArray())
//    {
//      b.append((char) ((int) c + (int) offshift));
//    }
//    return b.toString();
//  }
//  public String getPath(String file)
//  {
//    return ClassUtils.getSourceDirectory(getClass()).getAbsolutePath() + File.separator + file;
//  }
	}
}