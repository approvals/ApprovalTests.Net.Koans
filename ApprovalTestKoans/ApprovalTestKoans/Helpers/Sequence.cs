namespace ApprovalTestKoans.Helpers
{
	public class Sequence
	{
		public static string Print(int arg)
		{
			
		}
	}
}