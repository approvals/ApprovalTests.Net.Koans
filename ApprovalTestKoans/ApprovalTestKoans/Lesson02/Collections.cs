using System;
using System.Collections.Generic;
using ApprovalTestKoans.Helpers;
using ApprovalTests;
using ApprovalTests.Reporters;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace ApprovalTestKoans.Lesson02
{
	/**
 * How to do Koans:
 * Step 1: Press the Run Button (Place cursor on the Method name to run a single method)
 * PC: Ctrl+F11
 * Mac: Command+fn+F11
 * Step 2: Read the name of the Method that Failed
 * Step 3: Fill in the blank (___) to make it pass
 * Step 4: Repeat Until Enlightenment
 * Do not change anything except the blank (___)
 */

	[UseReporter(typeof (DiffReporter))]
	[TestClass]
	public class Collections : Koans
	{
		[TestMethod]
		public void SimpleArrays()
		{
			string[] approvalsIsPortedTo = {___, "Java", "Perl", "Ruby", "Php", "Node.js"};
			Approvals.VerifyAll(approvalsIsPortedTo, "Language");
		}

		[TestMethod]
  public void ListsWorkTheSameAsArrays() 
  {
    var approvalsIsGoodFor = new List<string>();
    approvalsIsGoodFor.Add("Strings");
    approvalsIsGoodFor.Add("Objects");
    approvalsIsGoodFor.Add("Log Files");
    approvalsIsGoodFor.Add("WinForms");
    approvalsIsGoodFor.Add("WPF");
    approvalsIsGoodFor.Add("RDLC Reports");
    approvalsIsGoodFor.Add("ASP Pages");
    approvalsIsGoodFor.Add("ASP.MVC Pages");
    approvalsIsGoodFor.Add("ASP Routes");
    approvalsIsGoodFor.Add("Entity Framework Queries");
    approvalsIsGoodFor.Add(___);
    approvalsIsGoodFor.Add("Xml");
    approvalsIsGoodFor.Add("Html");
    approvalsIsGoodFor.Add("Json");
    approvalsIsGoodFor.Add("Email");
    Approvals.VerifyAll( approvalsIsGoodFor,"Good For");
  }
		[TestMethod]
		public void DictionariesAreSortedBeforeDisplay()
  {
    var m = new Dictionary<int, string>();
    m[1000]= "M";
    m[10]= "X";
    m[1]= "I";
    m[100]= "C";
    m[5]= "V";
    m[500]= "D";
    m[50]= ___;
    Approvals.VerifyAll(m);
  }
		[TestMethod]
  public void TheUseOfTheLabel() 
  {
    string[] greatTv = {
        "Firefly", "Game of Throne", "Futurama", "Arrested Development", "Big Bang Theory", "Better Off Ted"};
    Approvals.VerifyAll( greatTv,___);
  }
//  @Test
//  public void ApprovalHelpShowWhenThingsHaveBeenRemoved() throws Exception
//  {
//    HashMap<String, String> m = new HashMap<String, String>();
//    m.put("Miss Scarlet", "Conservitory");
//    m.put("Colonel Mustard", "Kitchen");
//    m.put("Mrs. White", "Billiard Room");
//    m.put("Mr. Green", "Library");
//    m.put("Mrs. Peacock", "Dining Room");
//    m.put("Professor Plum", "Study");
//    m.remove(___);
//    Approvals.verify(m);
//  }
//  @Test
//  public void ApprovalHelpShowWhenThingsHaveBeenAdded() throws Exception
//  {
//    HashMap<String, String> punch = new HashMap<String, String>();
//    punch.put("Orange Juice", "2 cups");
//    punch.put("Lime Juice", "1/2 cup");
//    punch.put("Fruit Juice", "1 cup");
//    punch.put("Strawberries", "20");
//    punch.put("Mint leaves", "3");
//    punch.put("Sprite", "2 cups");
//    //punch.put(___,___); // <-- Uncomment Line to solve
//    Approvals.verify(punch);
//  }
//  @Test
//  public void HeadersHelpExplainTheContextWhenThingsAreConfusing() throws Exception
//  {
//    String header = ___;
//    Integer[] values = {1, 2, 6, 9, 11};
//    Approvals.verifyAll(header, values, new Sequence());
//  }
//  @Test
//  public void TransformingArraysManually() throws Exception
//  {
//    int[] numbers = {1, 3, ____, 10, 12, 16};
//    StringBuffer b = new StringBuffer();
//    for (int i : numbers)
//    {
//      b.append(String.format("%s^2 = %s\n", i, i * i));
//    }
//    Approvals.verify(b);
//  }
//  @Test
//  public void TransformingArraysWithFunctions() throws Exception
//  {
//    Integer[] numbers = {1, 3, 5, 10, 12, 16};
//    Approvals.verifyAll("Squares", numbers, new Function1<Integer, String>()
//    {
//      public String call(Integer i)
//      {
//        return ___;
//      }
//    });
//  }
//  @Test
//  public void TransformingArraysWithLambdas() throws Exception
//  {
//    // This requires java8
//    Integer[] numbers = {1, 3, 5, 10, 12, 16};
//    Approvals.verifyAll("Squares", numbers, n -> ___);
//  }
	}
}