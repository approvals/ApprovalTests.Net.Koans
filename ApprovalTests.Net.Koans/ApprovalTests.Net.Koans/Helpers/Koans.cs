using System.Text;

namespace ApprovalTestKoans.Helpers
{
	public class Koans
	{
		public string ___ = "Please fill in the blank";
		public int ____ = 0;

		public string decode(string text)
		{
			int offshift = 9;
			return push(text, offshift);
		}

		public string push(string text, int offshift)
		{
			var b = new StringBuilder();
			foreach (char c in text)
			{
				b.Append((char) (c + offshift));
			}
			return b.ToString();
		}
	}
}