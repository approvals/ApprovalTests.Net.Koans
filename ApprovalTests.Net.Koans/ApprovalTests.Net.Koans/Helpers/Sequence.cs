using ApprovalUtilities.Utilities;

namespace ApprovalTestKoans.Helpers
{
	public class Sequence
	{
		public static string Print(int number)
		{
			int n1 = 1;
			int n2 = 1;
			for (int i = 2; i <= number; i++)
			{
				int n3 = n1 + n2;
				n1 = n2;
				n2 = n3;
			}
			return $"#{number} => {n2}";
		}
	}
}