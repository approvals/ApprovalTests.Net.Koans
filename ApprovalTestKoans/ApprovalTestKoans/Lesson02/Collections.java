package org.approvaltests.koans.lesson02;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

import org.approvaltests.Approvals;
import org.approvaltests.koans.helpers.Koans;
import org.approvaltests.koans.helpers.Sequence;
import org.approvaltests.reporters.JunitReporter;
import org.approvaltests.reporters.UseReporter;
import org.junit.Test;
import org.lambda.functions.Function1;

/**
 * How to do Koans:
 * Step 1: Press the Run Button (Place cursor on the Method name to run a single method)
 * PC: Ctrl+F11
 * Mac: Command+fn+F11
 * Step 2: Read the name of the Method that Failed
 * Step 3: Fill in the blank (___) to make it pass
 * Step 4: Repeat Until Enlightenment
 * Do not change anything except the blank (___)
 */
@UseReporter(JunitReporter.class)
public class Collections extends Koans
{
  @Test
  public void SimpleArrays() throws Exception
  {
    String[] approvalsIsPortedTo = {".Net", ___, "Perl", "Ruby", "Php", "Node.js"};
    Approvals.verifyAll("Language", approvalsIsPortedTo);
  }
  @Test
  public void ListsWorkTheSameAsArrays() throws Exception
  {
    List<String> approvalsIsGoodFor = new ArrayList<String>();
    approvalsIsGoodFor.add("Strings");
    approvalsIsGoodFor.add("Objects");
    approvalsIsGoodFor.add("Log Files");
    approvalsIsGoodFor.add("JPanels");
    approvalsIsGoodFor.add(___);
    approvalsIsGoodFor.add("Xml");
    approvalsIsGoodFor.add("Html");
    approvalsIsGoodFor.add("Json");
    approvalsIsGoodFor.add("Email");
    Approvals.verifyAll("Good For", approvalsIsGoodFor);
  }
  @Test
  public void MapsAreSortedBeforeDisplay() throws Exception
  {
    HashMap<Integer, String> m = new HashMap<Integer, String>();
    m.put(1000, "M");
    m.put(10, "X");
    m.put(1, "I");
    m.put(100, "C");
    m.put(5, "V");
    m.put(500, "D");
    m.put(50, ___);
    Approvals.verify(m);
  }
  @Test
  public void TheUseOfTheLabel() throws Exception
  {
    String[] greatTv = {
        "Firefly", "Game of Throne", "Futurama", "Arrested Development", "Big Bang Theory", "Better Off Ted"};
    Approvals.verifyAll(___, greatTv);
  }
  @Test
  public void ApprovalHelpShowWhenThingsHaveBeenRemoved() throws Exception
  {
    HashMap<String, String> m = new HashMap<String, String>();
    m.put("Miss Scarlet", "Conservitory");
    m.put("Colonel Mustard", "Kitchen");
    m.put("Mrs. White", "Billiard Room");
    m.put("Mr. Green", "Library");
    m.put("Mrs. Peacock", "Dining Room");
    m.put("Professor Plum", "Study");
    m.remove(___);
    Approvals.verify(m);
  }
  @Test
  public void ApprovalHelpShowWhenThingsHaveBeenAdded() throws Exception
  {
    HashMap<String, String> punch = new HashMap<String, String>();
    punch.put("Orange Juice", "2 cups");
    punch.put("Lime Juice", "1/2 cup");
    punch.put("Fruit Juice", "1 cup");
    punch.put("Strawberries", "20");
    punch.put("Mint leaves", "3");
    punch.put("Sprite", "2 cups");
    //punch.put(___,___); // <-- Uncomment Line to solve
    Approvals.verify(punch);
  }
  @Test
  public void HeadersHelpExplainTheContextWhenThingsAreConfusing() throws Exception
  {
    String header = ___;
    Integer[] values = {1, 2, 6, 9, 11};
    Approvals.verifyAll(header, values, new Sequence());
  }
  @Test
  public void TransformingArraysManually() throws Exception
  {
    int[] numbers = {1, 3, ____, 10, 12, 16};
    StringBuffer b = new StringBuffer();
    for (int i : numbers)
    {
      b.append(String.format("%s^2 = %s\n", i, i * i));
    }
    Approvals.verify(b);
  }
  @Test
  public void TransformingArraysWithFunctions() throws Exception
  {
    Integer[] numbers = {1, 3, 5, 10, 12, 16};
    Approvals.verifyAll("Squares", numbers, new Function1<Integer, String>()
    {
      public String call(Integer i)
      {
        return ___;
      }
    });
  }
  @Test
  public void TransformingArraysWithLambdas() throws Exception
  {
    // This requires java8
    Integer[] numbers = {1, 3, 5, 10, 12, 16};
    Approvals.verifyAll("Squares", numbers, n -> ___);
  }
}
