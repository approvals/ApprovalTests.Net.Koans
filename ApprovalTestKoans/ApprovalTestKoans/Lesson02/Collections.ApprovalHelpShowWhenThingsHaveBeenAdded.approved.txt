Fruit Juice : 1 cup 
Lime Juice : 1/2 cup 
Mint leaves : 3 
Orange Juice : 2 cups 
Rum : 1/2 bottle 
Sprite : 2 cups 
Strawberries : 20 
